### ISSUE TITLE
curl http://npmjs.com/install.sh | sh

### ISSUE THREAD

**Issue by lotusirous** [npm/npm#17058]
I can not install npm by `curl http://npmjs.cominstall.sh | sh`.  The script does not create correct soft-link for npm

I installed node on `/home/trongkha/local`. However, the `install.sh` script show that the installation is complete.

```
Written by John Gilmore and Jay Fenlason.
install npm@latest
fetching: https://registry.npmjs.org/npm/-/npm-5.0.3.tgz
removed 267 packages in 20.316s
/home/trongkha/local/bin/npm -> /home/trongkha/local/lib/node_modules/npm/bin/npm-cli.js
+ npm@5.0.3
added 1 package in 2.667s
It worked
```
```
$ ls  /home/trongkha/local/lib/node_modules/npm/bin/npm-cli.js
ls: cannot access '/home/trongkha/local/lib/node_modules/npm/bin/npm-cli.js': No such file or directory
```
```
ls  /home/trongkha/local/lib/node_modules/ -al
total 8
drwxrwxr-x 2 trongkha trongkha 4096 Jun  7 07:06 .
drwxrwxr-x 3 trongkha trongkha 4096 Jun  7 07:06 ..
lrwxrwxrwx 1 trongkha trongkha   36 Jun  7 07:06 npm -> ../../../../../tmp/npm.17220/package
```

The directory ` ../../../../../tmp/npm.17220/package` does not exists. Thus, the soft-link does not link to the `npm-cli.js` script.

**Comment by anthonator** [comment]
I'm experiencing this as well

**Comment by chrmoritz** [comment]
This is a regression in the `install.sh` script (and the `make install` command) caused by npm@5's breaking change in `npm install <folder>` behaviour. See also #17036 and #16916 (point 2).

**Comment by apollo13** [comment]
Can we please reopen this issue? I am still seeing it :/

**Comment by develar** [comment]
For me solution was — use outdated npm that comes from node dist and recommend users to use reliable yarn instead of npm. Since in any case no reason to use npm if so amazing and really reliable tool as yarn exists.

**Comment by apollo13** [comment]
I worked around this by: manually downloading the tar.gz, run configure and make install and then replace the resulting symlink with the actual data.

**Comment by adrianheine** [comment]
Fixed in #17626.
